src/beautifiers/index.coffee

```diff
@@ -142,17 +142,16 @@ module.exports = class Beautifiers extends EventEmitter
 
   getLanguage : (grammar, filePath) ->
     # Get language
-    fileExtension = path.extname(filePath)
-    # Remove prefix "." (period) in fileExtension
-    fileExtension = fileExtension.substr(1)
-    languages = @languages.getLanguages({grammar, extension: fileExtension})
-    logger.verbose(languages, grammar, fileExtension)
-    # Check if unsupported language
-    if languages.length < 1
-      return null
-    else
-      # TODO: select appropriate language
-      language = languages[0]
+    if filePath
+      fileExtension = path.extname(filePath)
+      # Remove prefix "." (period) in fileExtension
+      fileExtension = fileExtension.substr(1)
+      languages = @languages.getLanguages({grammar, extension: fileExtension})
```

Comment:
`if filePath` conditional should change the `@langauges.getLanguages` call, however `grammar` is still useful for finding a language (even more important than `extension` actually).

Something like:

```
if filePath
    fileExtension = path.extname(filePath)
    fileExtension = fileExtension.substr(1)

if fileExtension
    languages = @languages.getLanguages({grammar, extension: fileExtension})
else
    languages = @languages.getLanguages({grammar})
```